src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt

```diff
@@ -52,13 +56,21 @@ sealed class RsDiagnostic(
         element: PsiElement,
         private val expectedTy: Ty,
         private val actualTy: Ty
-    ) : RsDiagnostic(element, inspectionClass = RsTypeCheckInspection::class.java) {
+    ) : RsDiagnostic(element, inspectionClass = RsTypeCheckInspection::class.java), TypeFoldable<TypeError> {
+        private val description: String? = if (expectedTy.hasTyInfer || actualTy.hasTyInfer) {
+            // if types contain infer types, they will be replaced with more specific (e.g. `{integer}` with `i32`)
+            // so we capture string representation of types right here
+            expectedFound(expectedTy, actualTy)
```

Comment:
Can you write a test for this case?

Reply:
Hmm, this should be covered by multiple tests, but we don't have any tests that check such error description (of any annotation), so if we want to start testing it, maybe we can do it in a separate PR?